rfcs/rfc0019.md

```diff
@@ -0,0 +1,174 @@
+are likely to keep hitting confusing problems with the behavior of `qq`.  We
+should add something that designs away all these problems now and for the
+future.
+
+## Rationale
+
+The proposed `qt` operator only looks for one special token in a string
+literal: `{`.  Source content until the matching `}` is treated as a scalar
+expression to be interpolated into the string in place of the `{...}`
+construct.  This is arbitrarily extensible for other existing expressions and
+for new ones that may be added.
+
+## Specification
+
+This:
+
+```
+qt{A { TEXT } B};
+```
+
+will be equivalent to
+
+```
+qq{A ${ \scalar TEXT } B};
+```
+
+To provide a heredoc version of qt, this:
+
+```
+<<qt{END}
+TEXT
+END
+```
+
+will apply qt-like interpolation to TEXT.  This (very rarely-seen) peril will
+exist in qt heredocs as it does in qq heredocs:
+
+```
+$str = <<"END";
+A ${
+END
+}
+END
+```
+
+(The peril here is that the first `END` terminates the string, meaning that the
+`${` is never closed.)
+
+## Backwards Compatibility
+
+Existing static analysis should be able to understand `qt` as a new quote-like
+operator fairly easily.  If the analyzer parses `${...}` within a `qq` string,
+parsing the `{...}` forms in a `qt` string should be possible.
+
+`qt` will need to be made available by a feature guard.  Use of the qt operator
+in source without the feature enabled would often parse as a subroutine call.
+
+There is prior art on the CPAN, most specifically in
+[Quote::Code](https://metacpan.org/pod/Quote::Code), which provides a `qc`.
+The syntax provided by Quote::Code is similar to, but not identical with, that
+proposed here.
+
+## Security Implications
+
+None identified at time of writing.
+
+## Examples
+
+The **Motivation** section, above, provides examples of what *doesn't* work
+well with existing interpolation.  Documentation can start with simple
+interpolation in both, then show how to use qt strings to allow more forms of
+interpolation.
+
+```
+# No interpolation
+qt{Greetings};
+
+# Simple scalar interpolation
+qt<Greetings, {$title} {$name}>;
+
+# Interpolation of method calls
+qt"Greetings, {$user->title} {$user->name}";
+
+# Interpolation of various expressions
+qt{It has been {$since{n}} {$since{units}} since your last login};
+
+qt{...a game of {join q{$"}, $favorites->{game}->name_words->@*}};
+```
+
+## Prototype Implementation
+
+Quote::Code, mentioned above, provides something very similar.
+
+## Future Scope
+
+`qt` is "qq but different".  If successful, it might suggest the usefulness of
+qt-like forms of qr, m, s, and other string-interpolating contexts.  This also
+suggests that another possible design for qt would be a pragma to change the
+behavior of qq *and other interpolating quote-like operators*.  This is more
+complex for the reader (because any given piece of code must be considered in
+terms of the enabled features), but eliminates the proliferation of QLOPs.
+
```

Comment:
Since we're considering adding this `qt` operator, and elsewhere there was vague mention of a `qa` or similar, I wonder if we want to declare as reserved any two-letter operator name beginning with `q`? E.g. by adding a warning like

```
$ perl -c "sub qz { ... }"
The identifier 'qz' is reserved for possible future expansion at -e line 1.
```